codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AwsEventStreamUtils.java

```diff
@@ -1878,7 +1882,7 @@ protected Void getDefault(Shape shape) {
 
         private void writeSetter(String valueTypeSymbolName) {
             var ds = CodegenUtils.getAsValueIfDereferencable(pointableIndex, memberShape, dataSource);
-            writer.write("$L.Set($S, $T($L))", target, headerName,
+            writer.write("$L.Headers.Set($S, $T($L))", target, headerName,
```

Comment:
Is this still valid if the event didn't only contain headers?

Reply:
This is valid, this snippet of code actually comes from a helper class within the `AwsEventStreamUtils` class that is responsible for setting headers.

Otherwise there is separate code after using this utility class that will handle the message payload.